### PR TITLE
rename delete blob snapshot options, add delete blob snapshot test cases

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp [hunk 1]
```diff
@@ -802,4 +802,22 @@ namespace Azure { namespace Storage { namespace Test {
...
 
+  TEST_F(BlockBlobClientTest, DeleteSnapshots)
+  {
+    auto blobClient = Azure::Storage::Blobs::BlockBlobClient::CreateFromConnectionString(
+        StandardStorageConnectionString(), m_containerName, RandomString());
+    auto s1 = blobClient.CreateSnapshot()->Snapshot;
+    Blobs::DeleteBlobOptions deleteOptions;
+    EXPECT_THROW(blobClient.Delete(deleteOptions), StorageException);
+    deleteOptions.DeleteSnapshots = Blobs::Models::DeleteSnapshotsOption::OnlySnapshots;
+    blobClient.Delete(deleteOptions);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:813]:
```suggestion
    EXPECT_NO_THROW(blobClient.Delete(deleteOptions));
```

File: sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp [hunk 2]
```diff
@@ -802,4 +802,22 @@ namespace Azure { namespace Storage { namespace Test {
...
+    EXPECT_NO_THROW(blobClient.GetProperties());
+    EXPECT_THROW(blobClient.WithSnapshot(s1).GetProperties(), StorageException);
+    auto s2 = blobClient.CreateSnapshot()->Snapshot;
+    deleteOptions.DeleteSnapshots = Blobs::Models::DeleteSnapshotsOption::IncludeSnapshots;
+    blobClient.Delete(deleteOptions);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:818]:
```suggestion
    EXPECT_NO_THROW(blobClient.Delete(deleteOptions));
```

File: sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp [hunk 3]
```diff
@@ -802,4 +802,22 @@ namespace Azure { namespace Storage { namespace Test {
     }
   }
 
+  TEST_F(BlockBlobClientTest, DeleteSnapshots)
+  {
+    auto blobClient = Azure::Storage::Blobs::BlockBlobClient::CreateFromConnectionString(
+        StandardStorageConnectionString(), m_containerName, RandomString());
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:808]:
```suggestion
    std::vector<uint8_t> emptyContent;
    auto blobClient = Azure::Storage::Blobs::BlockBlobClient::CreateFromConnectionString(
        StandardStorageConnectionString(), m_containerName, RandomString());
    blobClient.UploadFrom(emptyContent.data(), emptyContent.size());
```

File: sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp [hunk 4]
```diff
@@ -802,4 +802,24 @@ namespace Azure { namespace Storage { namespace Test {
...
+    EXPECT_THROW(blobClient.Delete(deleteOptions), StorageException);
+    deleteOptions.DeleteSnapshots = Blobs::Models::DeleteSnapshotsOption::OnlySnapshots;
+    EXPECT_NO_THROW(blobClient.Delete(deleteOptions));
+    EXPECT_NO_THROW(blobClient.GetProperties());
+    EXPECT_THROW(blobClient.WithSnapshot(s1).GetProperties(), StorageException);
+    auto s2 = blobClient.CreateSnapshot()->Snapshot;
+    deleteOptions.DeleteSnapshots = Blobs::Models::DeleteSnapshotsOption::IncludeSnapshots;
+    EXPECT_NO_THROW(blobClient.Delete(deleteOptions));
+    EXPECT_THROW(blobClient.GetProperties(), StorageException);
+    EXPECT_THROW(blobClient.WithSnapshot(s2), StorageException);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp:822]:
```suggestion
    EXPECT_THROW(blobClient.WithSnapshot(s2).GetProperties(), StorageException);
```

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp [hunk 5]
```diff
@@ -349,7 +349,7 @@ namespace Azure { namespace Storage { namespace Blobs {
       bool operator!=(const DeleteSnapshotsOption& other) const { return !(*this == other); }
       const std::string& Get() const { return m_value; }
       const static DeleteSnapshotsOption IncludeSnapshots;
-      const static DeleteSnapshotsOption Only;
+      const static DeleteSnapshotsOption OnlySnapshots;
```
Comment (RickWinter) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:352]: This naming is inconsistent with other languages.  I like what you've chosen better than other languages, but maybe there was a reason for their names.  

@tg-msft any opinion on this?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:352]: I made this change to align with .Net

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:352]: in .Net code:

```
#region enum DeleteSnapshotsOption
namespace Azure.Storage.Blobs.Models
{
    /// <summary>
    /// Required if the blob has associated snapshots. Specify one of the following two options: include: Delete the base blob and all of its snapshots. only: Delete only the blob's snapshots and not the blob itself
    /// </summary>
    public enum DeleteSnapshotsOption
    {
        /// <summary>
        /// none
        /// </summary>
        None,

        /// <summary>
        /// include
        /// </summary>
        IncludeSnapshots,

        /// <summary>
        /// only
        /// </summary>
        OnlySnapshots
    }
}
```

**Review by katmsft — APPROVED** [review]

**Pull request merged**